torch/_subclasses/fake_tensor.py

```diff
@@ -901,6 +931,25 @@ class TensorMetadata:
     dense_dim: Optional[int]
     sparse_dim: Optional[int]
 
+    def flatten_into(self, result: List[object]) -> None:
```

Comment:
would this be simpler as :
```
for field in dataclasses.fields(self):
    result.append(getattr(self, field)
```

Reply:
drive-by comment: cant we also implement slots prior 3.10 using `__slots__ = dataclasses.fields(self)`?

Reply:
> would this be simpler as :
Neat! I didn't know about that.

> drive-by comment: cant we also implement slots prior 3.10 using __slots__ = dataclasses.fields(self)

You can't do it inside the class because `self`, `__class__`, `THE_ACTUAL_CLASSNAME` isn't defined yet.  I also tried:
```
X.__slots__ = tuple(f.name for f in dataclasses.fields(X))
```
which looks like it works... but... I wrote a microbenchmark to test.  Without slots at all the benchmark takes 1.84s.  With a manually written `__slots__` it takes 1.59s.  Adding slots after the class is defined takes 1.85s. So it seems that there's some magic to defining the slots within the class definition.

And in fact looking at the source code for dataclasses.py it says:
```
# Need to create a new class, since we can't set __slots__
#  after a class has been created.
```
and they do some serious magic to create the new class. Hm. I suppose we could steal their code as a "backport" and put it in utils...